ServerCore/SyncController.cs

```diff
@@ -0,0 +1,534 @@
+        ///   to the page of the corresponding puzzle.
+        /// </summary>
+        private string GetSiteVersion(Puzzle puzzle)
+        {
+            // if (puzzle.ID == 8) { return "2" };   // use a line like this to indicate a puzzle-specific version
+            // if (puzzle.ID == 97) { return "3" };  // use a line like this to indicate a puzzle-specific version
+            return "1";
+        }
+
+        /// <summary>
+        ///   This routine handles sync request aspects that require fetching a list of all the puzzles the team
+        ///   has solved.
+        /// </summary>
+        private async Task HandleSyncAspectsRequiringListOfSolvedPuzzles(DecodedSyncRequest request, SyncResponse response,
+                                                                         string groupExcludedFromSolveCount, int teamId, int eventId)
+        {
+            // If the requester isn't asking for pieces (by setting MinSolveCount to null), and isn't asking about
+            // whether any puzzle IDs are solved, we can save time by not querying the list of solved puzzles.
+
+            if (request.MinSolveCount == null && (request.QueryPuzzleIds == null || request.QueryPuzzleIds.Count == 0)) {
+                return;
+            }
+
+            // If the request is asking which of the puzzle IDs in request.QueryPuzzleIds has been
+            // solved, create a HashSet so that we can quickly look up if an ID is in that list.
+
+            HashSet<int> queryPuzzleIdSet = null;
+            if (request.QueryPuzzleIds != null) {
+                queryPuzzleIdSet = new HashSet<int>();
+                foreach (var queryPuzzleId in request.QueryPuzzleIds) {
+                    queryPuzzleIdSet.Add(queryPuzzleId);
+                }
+            }
+
+            // Get a list of all the puzzles this team has solved from this event.
+
+            List<Puzzle> solves = await (from state in context.PuzzleStatePerTeam
+                                         where state.TeamID == teamId && state.SolvedTime != null
+                                         select state.Puzzle).ToListAsync();
+
+            int maxSolveCount = 0;
+            List<int> solvedPuzzles = new List<int>();
+            foreach (var solvedPuzzle in solves) {
+                // I'm pretty sure the query above will never return any puzzles that aren't
+                // from the current event.  But, just in case, out of paranoia, we check
+                // for them and don't count them.
+
+                if (solvedPuzzle.Event.ID != eventId) {
+                    continue;
+                }
+
+                // If the request is asking whether certain puzzles are solved, check if it's
+                // in the set and, if so, put it in the list of solved puzzles to inform the
+                // requester of.
+                    
+                if (queryPuzzleIdSet != null && queryPuzzleIdSet.Contains(solvedPuzzle.ID)) {
```

Comment:
solveValue >= 10 check here?

Reply:
All that we're doing here is responding to a question the requester has about which puzzles have been solved.  For instance, the requester might want to know if a certain Fast Forward was taken, and that Fast Forward might have a solve value under 10.  So we want the requester to be able to ask such questions even about puzzles whose solve value is under 10.